src/main/java/com/example/ShortnerURL/service/ApiService.java

```diff
@@ -0,0 +1,99 @@
+
+
+import java.net.URL;
+import java.util.List;
+import java.util.Map;
+
+@Service
+@RequiredArgsConstructor
+public class ApiService {
+    private final ShortLinkRepository shortLinkRepository;
+    private final ModelMapper modelMapper;
+    private static final String DOMAIN = "example.com/";
+    private final Map<String, String> urlMapping;
+
+
+    public List<ShortLinkDto> getAllShortLinks() {
+        return shortLinkRepository.findAll()
+                .stream()
+                .map(ApiService::convertToDto)
+                .toList();
+    }
+
+    private static ShortLinkDto convertToDto(ShortLinkEntity shortLinkEntity) {
+        return ShortLinkDto.builder()
+                .shortLinkCode(shortLinkEntity.getShortLinkCode())
+                .shortLink(DOMAIN + shortLinkEntity.getShortLinkCode())
+                .longLink(shortLinkEntity.getLongLink())
+                .build();
+    }
+
+    @Transactional
+    public void deleteShortLink(String shortLinkCode) throws ShortLinkNotFoundException {
+        if (!shortLinkRepository.existsByShortLinkCode(shortLinkCode)) {
+            throw new ShortLinkNotFoundException(shortLinkCode);
+        }
+        shortLinkRepository.deleteByShortLinkCode(shortLinkCode);
+    }
+
+
+    public ShortLinkDto getShortLink(String shortLinkCode) {
+
+        return shortLinkRepository.findByShortLinkCode(shortLinkCode)
+                .stream()
+                .findFirst()
+                .map(ApiService::convertToDto)
+                .orElseThrow(() -> new ShortLinkNotFoundException(shortLinkCode));
+    }
+
+    public ShortLinkDto createShortLink(String longLink) throws ApplicationExceptions {
+        String shortLinkCode = "";
+        if (!validateURL(longLink)) {
+            throw new InvalidUrlException(longLink);
+        }
+        do {
+            shortLinkCode = RandomStringUtils.randomAlphanumeric(6);
+        } while (shortLinkRepository.existsByShortLinkCode(shortLinkCode));
+
+        ShortLinkEntity shortLinkEntity = ShortLinkEntity.builder()
+                .shortLinkCode(shortLinkCode)
+                .longLink(longLink)
+                .build();
+        ShortLinkEntity savedEntity = shortLinkRepository.save(shortLinkEntity);
+        return convertToDto(savedEntity);
+    }
+
+    private static boolean validateURL(String longLink) {
+        try {
+            new URL(longLink).toURI();
+        } catch (Exception e) {
+            System.out.println(e);
+            return false;
+        }
+        return true;
+    }
+
+    public RedirectView redirectToOriginalUrl(String shortLinkCode) throws ShortLinkNotFoundException{
+        List<ShortLinkEntity> shortLinkEntities = shortLinkRepository.findByShortLinkCode(shortLinkCode);
+        ShortLinkEntity shortLinkEntity = shortLinkEntities
+                .stream()
+                .findFirst()
+                .orElseThrow(() -> new ShortLinkNotFoundException(shortLinkCode));
```

Comment:
instead of raising a not found exception, we would need to redirect to an actual page that tells the user that link has not been found.